### ISSUE TITLE
Could not find module ember-get-config imported from (require)

### ISSUE THREAD

**Issue by maxymczech** [embroider-build/embroider#823]
Hello. I am trying to migrate Ember app to Embroider and I am getting this error from several addon packages:

- `ember-cli-moment-shim`
- `ember-drag-drop-polyfill`

I am using Ember 3.26 and Embroider 0.41.0

**Comment by maxymczech** [comment]
@alexlafroscia it seems this error might be caused by `ember-get-config` compat adapter that you wrote in #770. I have tried removing it from my `node_modules/@embroider/compat/src/ember-get-config.*` and now my application loads perfectly fine. What could be the reason for that? If you have time, can you please look into this?

**Comment by ef4** [comment]
It’s possible to disable a built in compat adapter if it’s causing trouble.
You set the `compatAdapters` option to a Map where the name of the addon
maps to null.

**Comment by maxymczech** [comment]
@ef4 Thank you very much, it works. The question however remains, why did the compat adapter cause problems in the first place.

**Comment by ef4** [comment]
Agreed, that is just a workaround until we can sort it out. If it's not needed, great, we can delete it. The goal is to eventually delete all the compat adapters as addons become compatible enough without them.

**Comment by maxymczech** [comment]
Thank you for clarification. In that case I am closing the issue.

**Comment by ef4** [comment]
We can keep this open so we remember to either delete it or figure out which versions really need it.

**Comment by alexlafroscia** [comment]
The compat adapter was created originally because `ember-get-config` generates (or did, at the time that I wrote it) a module that _itself_ depends on your host app without actually listing it as a dependency. When talking through that fact with @ef4, it was suggested that we use a compat adapter to generate a different implementation with the same outcome but an implementation that doesn't require the addon depending on the host app.

What kind of error did you see?

**Comment by maxymczech** [comment]
> What kind of error did you see?

The one in issue name: `Could not find module ember-get-config imported from (require)`

**Comment by johanrd** [comment]
I get the same error as @maxymczech above when running with `v > 0.41` (when e.g. `ember-inline-edit` is installed). No errors when building with `v0.40`.

Adding `compatAdapters: new Map([['ember-get-config', null]])` did not seem to help, so I'm stranded on `v0.40` for now.

**Comment by maxymczech** [comment]
@johanrd Our app is currently on version `0.43.0` of embroider. Here is the config that is currently working for me, maybe something from there will help:

```
  return require('@embroider/compat').compatBuild(app, Webpack, {
    compatAdapters: new Map([
      ['ember-get-config', null]
    ]),

    packagerOptions: {
      webpackConfig: {
        // Generate source maps
        devtool: (process.env.EMBER_ENV === 'production' ? 'cheap-source-map' : 'eval'),

        plugins: [
          new MomentLocalesPlugin({
            localesToKeep: ['cs']
          })
        ],

        module: {
          rules: [{
            test: /ember-concurrency/,
            loader: 'string-replace-loader',
            options: {
              multiple: [
                {
                  search: '\\[yieldableSymbol\\]',
                  flags: 'g',
                  replace: '["__ec_yieldable__"]'
                },
                {
                  search: '\\[cancelableSymbol\\]',
                  flags: 'g',
                  replace: '["__ec_cancel__"]'
                }
              ]
            }
          }]
        }
      }
    },
    splitAtRoutes: [/^((?!application).)*$/],
    staticAddonTestSupportTrees: true,
    staticAddonTrees: true,
    staticComponents: true,
    staticHelpers: true
  });
```

**Comment by johanrd** [comment]
FYI: Tried to reproduce now, and the issue comes back if I regenerate the `yarn.lock` file from scratch. Not sure what pin change is causing it, though.

**Comment by rahulk94** [comment]
>FYI: Tried to reproduce now, and the issue comes back if I regenerate the yarn.lock file from scratch. Thus, the fix seems a bit vulnerable. Not sure what pin change is causing it, though.

I also had things working previously with the existing `ember-get-config` compat adapter but after rebasing my branch onto master and regenerating my lock file with changes for Embroider this issue has come back for me. Setting the compat adapter to null also doesn't help unfortunately.

**Comment by ef4** [comment]
This issue is fixed in PR #1030, which isn't merged yet.
